packages/runtime-core/src/components/Teleport.ts

```diff
@@ -215,8 +215,9 @@ export const TeleportImpl = {
     vnode: VNode,
     { r: remove, o: { remove: hostRemove } }: RendererInternals
   ) {
-    const { shapeFlag, children, anchor } = vnode
+    const { shapeFlag, children, anchor, targetAnchor } = vnode
     hostRemove(anchor!)
+    hostRemove(targetAnchor!)
```

Comment:
Here should check the `target` existence, because it only insert when has vaild `target`.